pkg/kube/volume/volume.go

```diff
@@ -113,21 +113,25 @@ type CreatePVCFromSnapshotArgs struct {
 // CreatePVCFromSnapshot will restore a volume and returns the resulting
 // PersistentVolumeClaim and any error that happened in the process.
 func CreatePVCFromSnapshot(ctx context.Context, args *CreatePVCFromSnapshotArgs) (string, error) {
-	sns, err := snapshot.NewSnapshotter(args.KubeCli, args.DynCli)
-	if err != nil {
-		return "", err
-	}
-	snap, err := sns.Get(ctx, args.SnapshotName, args.Namespace)
-	if err != nil {
-		return "", err
-	}
-	size := snap.Status.RestoreSize
-	if args.RestoreSize != nil {
+	var size *resource.Quantity
+	if args.RestoreSize == nil {
+		sns, err := snapshot.NewSnapshotter(args.KubeCli, args.DynCli)
+		if err != nil {
+			return "", err
+		}
+		snap, err := sns.Get(ctx, args.SnapshotName, args.Namespace)
+		if err != nil {
+			return "", err
+		}
+
+		size = snap.Status.RestoreSize
+	} else {
 		s := resource.MustParse(fmt.Sprintf("%dGi", *args.RestoreSize))
```

Comment:
Merging this - @PrasadG193 - in a followup can we switch to `resource.ParseQuantity` instead of `MustParse`?

Reply:
Sounds good!